get_http.go

```diff
@@ -57,13 +61,17 @@ func (g *HttpGetter) Get(dst string, u *url.URL) error {
 		}
 	}
 
+	if g.Client == nil {
+		g.Client = http.DefaultClient
```

Comment:
This seems a little odd, what is the reason to default to the `http.DefaultClient` here? If it wasn't set it seems reasonable to use a fresh client from `cleanhttp`. Am I missing something?

Reply:
This came from the first commit before cleanhttp, where the Client field was added, and it needed a default. I left it to preserve the existing behavior of a user making a new HTTPGetter and expecting it to inherit the `http.DefaultClient`, which is also documented on the struct field.

If you want to excise the old `http.DefaultClient` entirely, I can make a single global http client for this case, but we don't want to insert a new client every time we hit this because it will leak http connections.

Reply:
Agreed it should be a private package global. IMO it's better to replace the client using the public `Getters` map on a `*getter.Client` if a custom client is needed. I think it's nicer to just use `cleanhttp` by default. We do the same in other packages like `go-retryablehttp`.